java/src/jmri/jmrix/loconet/loconetovertcp/LnOverTcpPacketizer.java

```diff
@@ -109,9 +109,14 @@ public RcvHandler(LnOverTcpPacketizer lt) {
         public void run() {
 
             String rxLine;
-            while (true) {  // loop permanently, program close will exit
+            while (! Thread.interrupted()) {  // loop until interrupted in dispose() call
                 try {
                     // start by looking for a complete line
+                    
+                    // Note:  the readline() call blocks and does not respond to the
+                    // thread being interrupted.  A join() on this thread will block
+                    // permanently.  This needs to be re-written to check for
+                    // available content before doing a read() from the stream.
                     rxLine = istream.readLine();
```

Comment:
Just a question:
I assume that you mean that it isn't enough to just check if there is any content to the stream, but that the readline() call must be changed to read() calls. If there is content on the stream but not a newline on the stream, I assume the readline() will block anyway.